graphs/check_bipartite_graph_dfs.py

```diff
@@ -1,34 +1,50 @@
-# Check whether Graph is Bipartite or Not using DFS
-
-
-# A Bipartite Graph is a graph whose vertices can be divided into two independent sets,
-# U and V such that every edge (u, v) either connects a vertex from U to V or a vertex
-# from V to U. In other words, for every edge (u, v), either u belongs to U and v to V,
-# or u belongs to V and v to U. We can also say that there is no edge that connects
-# vertices of same set.
-def check_bipartite_dfs(graph):
-    visited = [False] * len(graph)
-    color = [-1] * len(graph)
-
-    def dfs(v, c):
-        visited[v] = True
-        color[v] = c
-        for u in graph[v]:
-            if not visited[u]:
-                dfs(u, 1 - c)
-
-    for i in range(len(graph)):
-        if not visited[i]:
-            dfs(i, 0)
-
-    for i in range(len(graph)):
-        for j in graph[i]:
-            if color[i] == color[j]:
+from collections import defaultdict
+
+
+def is_bipartite(graph: defaultdict[int, list[int]]) -> bool:
+    """
+    Check whether a graph is Bipartite or not using Depth-First Search (DFS).
+
+    A Bipartite Graph is a graph whose vertices can be divided into two independent
+    sets, U and V such that every edge (u, v) either connects a vertex from
+    U to V or a vertex from V to U. In other words, for every edge (u, v),
+    either u belongs to U and v to V, or u belongs to V and v to U. There is
+    no edge that connects vertices of the same set.
+
+    Args:
+        graph (DefaultDict[int, List[int]]): An adjacency list representing the graph.
```

Comment:
Please do not repeat the datatypes which are already stated in the function signature and verified by mypy.  Readers become confused if one instance is changed and the other is not.  Please use the additional space to define what an adjacency list is.
```suggestion
        graph: An adjacency list representing the graph.
```